src/plugins/graph/models/graph-model.ts

```diff
@@ -461,25 +461,20 @@ export const GraphModel = TileContentModel
       }
 
       // Display a plotted function when this is linked to a SharedVariableModel
-      addDisposer(self, reaction(
-        () => {
-          const smm = getSharedModelManager(self);
-          let sharedVariableModels;
-          if (smm?.isReady) {
-            sharedVariableModels = smm.getTileSharedModelsByType(self, SharedVariables);
-          }
-          return sharedVariableModels;
-        },
-        (sharedVariableModels) => {
-          if (sharedVariableModels && sharedVariableModels.length > 0) {
-            const plottedFunctionAdornment = PlottedFunctionAdornmentModel.create();
-            plottedFunctionAdornment.addPlottedFunction(x => x**2);
-            self.showAdornment(plottedFunctionAdornment);
-          } else {
-            self.hideAdornment(kPlottedFunctionType);
-          }
+      addDisposer(self, autorun(() => {
+        const smm = getSharedModelManager(self);
+        let sharedVariableModels;
+        if (smm?.isReady) {
+          sharedVariableModels = smm.getTileSharedModelsByType(self, SharedVariables);
+        }
+        if (sharedVariableModels && sharedVariableModels.length > 0) {
+          const plottedFunctionAdornment = PlottedFunctionAdornmentModel.create();
```

Comment:
This code was a bit wonky. I've made it so it looks for an existing adornment and tries to show that if it can find one, rather than creating a new one every time. I'm not sure if creating a new one every time was causing problems or not, but the new way is certainly clean. This code has also been moved into `updateAfterSharedModelChanges` instead of being in its own separate `autorun`.

In response to Boris's PR feedback I made a lot of changes that move us towards supporting multiple plots, and it does so with a single adornment with multiple variable plot instances (similar to how the original `PlottedFunctionAdornment` can have multiple `PlottedFunctionInstances`, each with its own function). I'm not sure if there are major advantages or disadvantages to supporting multiple plots via multiple adornments vs multiple instances, but at this point it would definitely take a lot less work to get the multiple instances system up and running.